### ISSUE TITLE
A way to define complex cross table constraints is needed

### ISSUE THREAD

**Issue by victordombrovskiy** [prisma/specs#67]
Example:

```
model Category {
  id                    ID
  value                 String      @unique
  types                 Type[]
}

// @unique(category.value, value) <-- something like this is needed
model Type {
  id                    ID
  value                 String
  category              Category
}
```

Type should be unique by `type.value`, but across a category. In other words, there may be several types with the same `value` if they belong to different categories.

**Comment by matthewmueller** [comment]
@victordombrovskiy thanks for bringing this up! I haven't considered this use case before. 

Is this possible in SQL? I've done a quick search and it doesn't seem like it's natively built-in.

**Comment by janpio** [comment]
Remind @victordombrovskiy 🐰

**Comment by victordombrovskiy** [comment]
You are right, looks like it's natively impossible. Anyway, sounds like a useful feature. Probably, you could think about supporting of it somehow on a higher level comparing to a native support.

**Comment by matthewmueller** [comment]
Note: unassigning myself on things I'm not actively working on. I may pick this up in the future.
